### ISSUE TITLE
LE 2.0.2 Rel 29 - nginx permission errors trying to reference Let's Encrypt certificates within Plesk Onyx nginx additional directives

### ISSUE THREAD

**Issue by amavarick** [plesk/letsencrypt-plesk#161]
I have the latest LE Extension installed 2.0.2 Rel 29.  Below is an excerpt from my /var/log/nginx/error.log.  I tried to enable OCSP stapling by adding some entries into the nginx additional directives.  **It allowed me to apply and ok without error but when I restarted nginx and the service wouldn't start till I commented out each and every entry that listed a certificate location.**  Adding these locations is relatively new, is there a possibility that permissions are wrong so Plesk/nginx cannot read?  Am I incorrectly using the full directory location rather than say /etc/live/mydomain.tld?  Why are the permissions so broad to the certificates?

**I do have question regarding the broad permission set granted**?
-rw-r--r--. 1 psaadm psaadm 423 Apr  6 10:36 README
**lrwxrwxrwx**. 1 psaadm psaadm  42 Apr  6 10:36 privkey.pem -> ../../archive/domain.tld/privkey1.pem
**lrwxrwxrwx**. 1 psaadm psaadm  44 Apr  6 10:36 fullchain.pem -> ../../archive/domain.tld/fullchain1.pem
**lrwxrwxrwx**. 1 psaadm psaadm  40 Apr  6 10:36 chain.pem -> ../../archive/domain.tld/chain1.pem
**lrwxrwxrwx**. 1 psaadm psaadm  39 Apr  6 10:36 cert.pem -> ../../archive/domain.tld/cert1.pem

/var/log/nginx/error.log entry.  This is just a single entry of similar permission issue to just the let's encrypt certificates:
2017/04/06 11:16:24 [emerg] 9790#0: SSL_CTX_load_verify_locations("/usr/local/psa/var/modules/letsencrypt/etc/live/mydomain.tld/chain.pem") failed (SSL: error:0200100D:system library:fopen:**Permission denied:fopen('/usr/local/psa/var/modules/letsencrypt/etc/live/mydomain.tld/chain.pem','r')** error:2006D002:BI O routines:BIO_new_file:system lib error:0B084002:x509 certificate routines:X509 _load_cert_crl_file:system lib)

**Comment by rkosolapov** [comment]
> I tried to enable OCSP stapling by adding some entries into the nginx additional directives.

Hello.  Could you provide us with these entries?
And another question - did it work on 1.9?

**Comment by amavarick** [comment]
The entries are:

> ssl_stapling on;
> ssl_stapling_verify on;
> ssl_trusted_certificate /usr/local/psa/var/modules/letsencrypt/etc/live/domain.tld/chain.pem;
> resolver 8.8.4.4 8.8.8.8 valid=300s;
> resolver_timeout 10s;

This did work in an earlier version, I cannot say which version.

As a test, I can copy out the chain.pem and place it into /etc/ssl/stapling/chain.pem and then add the entries above just a different folder...

> ssl_stapling on;
> ssl_stapling_verify on;
> ssl_trusted_certificate /etc/ssl/stapling/chain.pem;
> resolver 8.8.4.4 8.8.8.8 valid=300s;
> resolver_timeout 10s;

This works just fine and I can validate that stapling works.  So there is certainly some sort of permissions issue when nginx tries to access the chain.pem file that does not occur in other directories...

> Permission denied:fopen('/usr/local/psa/var/modules/letsencrypt/etc/live/mydomain.tld/chain.pem','r')

**Comment by UFHH01** [comment]
Already discussed and investigated at:

=> https://talk.plesk.com/threads/ocsp-stapling-with-nginx-issue.339339/

**Comment by amavarick** [comment]
The actual issue is: SELinux is stopping the ability for nginx to access the /usr/local/psa/var/modules/letsencrypt/etc/ directory.

**Comment by amavarick** [comment]
Update - Plesk technical support is taking a while at this as they keep uncommenting the pem file and testing and saying everything works without trying to restart nginx service.  Setting OCSP and clicking apply or ok works without error.  It isn't till you try to restart nginx that you receive errors where SELinux will not allow the nginx service to start.

I setup several test Plesk/Onyx servers to test this out (Ubuntu, CentOS, Debian)  I cannot duplicate the issue with SELINUX or AppGuard enforcing.

I do not make back-end changes to my production box and am convinced this issue is from either an upgrade from Plesk to Onyx or the multiple updates for Let's Encrypt extension.

Wish other people who have gone through the multiple Let's Encrypt updates could test OCSP settings AND restart nginx service to see if they have the same issue.

**Comment by amavarick** [comment]
In working with Plesk support they were able to resolve the issue with SELINUX permissions blocking the LE certificate directory.  We were unable to replicate the problem on brand new systems.  Although I am convinced this is an issue from the LE removal and re-adding of the directory, I cannot replicate the problem so am closing this ticket.  We also identified that we do not have to identify the certificate as Plesk already identifies the certificate.  The following command is fine to enable OCSP.  Note: ssl_trusted_certificate commented out.  Just add the following to nginx additional directives:

#Enable OCSP
ssl_stapling on;
ssl_stapling_verify on;
#ssl_trusted_certificate /usr/local/psa/var/modules/letsencrypt/etc/live/yourdomain.tld/chain.pem;
resolver 8.8.4.4 8.8.8.8 valid=300s;
resolver_timeout 10s;

**Comment by rkosolapov** [comment]
amaverick,
just to be sure - your brand new systems have selinux activated, isn't it?

**Comment by amavarick** [comment]
I tested CentOS, Debian and Ubuntu (All using latest production/LTS 64 bit versions created on digitalocean droplets).  Before installing Plesk Onyx I ensured for CentOS and Debian that selinux was activated and enforcing.  For Ubuntu, I used AppArmor and ensured that all profiles were set to enforce.

For what its worth, I'm convinced there either was a bug from LE extension updates removing the directory and re-adding it, possibly not reconfiguring selinux profile... problem is that new systems don't reflect that and the SR Plesk support was unable to duplicate...Not sure they wanted to go through and install each version of LE to truly duplicate the problem.  There doesn't seem to be others with the same issue so I'll chalk this up to me being the only one.  I used this as an excuse to create a virgin hardened OS and install new Onyx stable version.
